sunpy/visualization/animator/image.py

```diff
@@ -212,6 +213,14 @@ def __init__(self, data, wcs=None, image_axes=[-1, -2], unit_x_axis=None, unit_y
         list_slices_wcsaxes[image_axes[0]] = 'x'
         list_slices_wcsaxes[image_axes[1]] = 'y'
         self.slices_wcsaxes = list_slices_wcsaxes[::-1]
+        if(unit_x_axis is not None):
```

Comment:
I believe I made this comment before, I really don't like this API. I feel like front-loading the constructor with the unit is the top of a slippery slope.

Reply:
ok, for the time being I am reverting that portion to way it was before, until I get a better solution